q.js

```diff
@@ -91,6 +91,7 @@ var nextTick =(function () {
     var flushing = false;
     var requestTick = void 0;
     var isNodeJS = false;
+    var laterQueue = []; // queue for late tasks
```

Comment:
In general I agree that two tiers are something that isn't enough - however in this particular case only the library calls `runAfter`.

While I agree a heap implementing a priority queue or some kind of chain of responsibility pattern is preferred I didn't want to bloat Q with too much code and wanted to add a single extra tier in the simplest way possible. I used an array instead of a linked list since it is much faster and less code to write.